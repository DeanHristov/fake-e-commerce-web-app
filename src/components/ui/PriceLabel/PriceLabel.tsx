import { FC } from 'react';
import { Utils } from '@/utils/Utils';
import { TCurrency } from '@/types';

export interface IPriceLabelProps {
  price: number;
  currency: TCurrency;
  discountPercentage?: number;
}

const PriceLabel: FC<IPriceLabelProps> = ({
  price,
  discountPercentage,
  currency = 'EUR',
}) => {
  return (
    <div
      className={`${Utils.isNotNull(discountPercentage) ? 'space-x-2' : ''}`}
    >
      {Utils.isNotNull(discountPercentage) && (
        <>
          <span className="price-label-old line-through text-sm">
            {Utils.calculateOldPrice(price, discountPercentage!, currency)}
          </span>
          <span className="price-label-old">{'/'}</span>
        </>
      )}
      <span className="price-label">
        {Utils.parseAmountByCurrency(price, currency)}
      </span>
    </div>
  );
};

export default PriceLabel;
