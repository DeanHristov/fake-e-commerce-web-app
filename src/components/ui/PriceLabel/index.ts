import PriceLabel, { IPriceLabelProps } from './PriceLabel';

export type { IPriceLabelProps };
export default PriceLabel;
