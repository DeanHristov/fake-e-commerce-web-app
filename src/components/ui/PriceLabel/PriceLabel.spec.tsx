import { render } from '@testing-library/react';
import PriceLabel from './';

describe('UI/Component <PriceLabel {...} />', () => {
  it('Should render the default state', () => {
    const { container } = render(<PriceLabel currency="EUR" price={34.2} />);

    expect(container).toMatchSnapshot();
  });

  it('Should render the both prices', () => {
    const { container } = render(
      <PriceLabel discountPercentage={12.9} price={34.2} currency={'USD'} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('Should be able to see the old price based on discount percentage', () => {
    const { getByText } = render(
      <PriceLabel currency="EUR" price={549} discountPercentage={12.9} />,
    );

    expect(getByText(/â‚¬620.00/i)).toBeVisible();
  });
});
