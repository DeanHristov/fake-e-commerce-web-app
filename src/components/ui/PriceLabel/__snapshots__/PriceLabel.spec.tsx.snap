// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Component <PriceLabel {...} /> Should render the both prices 1`] = `
<div>
  <div
    class="space-x-2"
  >
    <span
      class="price-label-old line-through text-sm"
    >
      $38.61
    </span>
    <span
      class="price-label-old"
    >
      /
    </span>
    <span
      class="price-label"
    >
      $34.20
    </span>
  </div>
</div>
`;

exports[`UI/Component <PriceLabel {...} /> Should render the default state 1`] = `
<div>
  <div
    class=""
  >
    <span
      class="price-label"
    >
      â‚¬34.20
    </span>
  </div>
</div>
`;
